cartographer/io/pcd_writing_points_processor.cc

```diff
@@ -32,20 +32,25 @@ namespace {
 
 // Writes the PCD header claiming 'num_points' will follow it into
 // 'output_file'.
-void WriteBinaryPcdHeader(const bool has_color, const int64 num_points,
-                          FileWriter* const file_writer) {
+void WriteBinaryPcdHeader(const bool has_color, const bool has_intensities,
+                          const int64 num_points, FileWriter* const file_writer) {
   std::string color_header_field = !has_color ? "" : " rgb";
   std::string color_header_type = !has_color ? "" : " U";
   std::string color_header_size = !has_color ? "" : " 4";
   std::string color_header_count = !has_color ? "" : " 1";
 
+  std::string intensity_header_field = !has_intensities ? "" : " intensity";
+  std::string intensity_header_type = !has_intensities ? "" : " F";
+  std::string intensity_header_size = !has_intensities ? "" : " 4";
```

Comment:
nit: these could all be made `const std::string...`. Same above for the color header strings.

Reply:
Yeah, you are right.